logstash-core/src/main/java/org/logstash/ackedqueue/io/MmapPageIO.java

```diff
@@ -50,17 +51,53 @@
 
     private MappedByteBuffer buffer;
 
-    public MmapPageIO(int pageNum, int capacity, Path dirPath) {
+    /**
+     * Constructor that sets the page num, the minimum size of the file, and its path
+     * If the file on disk already exists, and is larger than the minimum capacity, that size
+     * will be used instead.
+     * @param pageNum
+     * @param minimumFileSize
+     * @param dirPath
+     */
+    public MmapPageIO(int pageNum, int minimumFileSize, Path dirPath) {
         this.minSeqNum = 0;
         this.elementCount = 0;
         this.version = 0;
         this.head = 0;
-        this.capacity = capacity;
         this.offsetMap = new IntVector();
         this.checkSummer = new CRC32();
         this.file = dirPath.resolve("page." + pageNum).toFile();
+        int existingFileSize = (int) file.length();
```

Comment:
@andrewvc this is somewhat optimistic I think. We should do a safe cast here (using Guava's `Ints` for example) here and hard fail if the length exceeds 2GB because we can't `mmap` that kind of file.
I know what we won't be able to write that kind of file in the first place, so this shouldn't ever come up but with the amount of weird bugs we ran into I think we should have visibility on that kind of problem if it occurs :)

Reply:
+1 to handling the edge cases.